app/modules/users/permissions/rules.py

```diff
@@ -350,7 +323,28 @@ def _permitted_via_user(self, user):
     #             project_index = project_index + 1
 
     def _permitted_via_collaboration(self, user):
-        # TODO:
+        tried_users = [user]
+        object_user_methods = OBJECT_USER_METHOD_MAP.get(
+            (self._obj.__class__.__name__, self._action)
+        )
+
+        for collab_assoc in user.user_collaboration_associations:
+            collab_users = collab_assoc.collaboration.get_users()
+            for other_user in collab_users:
+                if other_user not in tried_users:
+                    tried_users.append(other_user)
+                    if other_user.owns_object(self._obj):
```

Comment:
Note this will need to be a bit more nuanced and will need to check what permission is actually granted.
At the moment this PR is creating the framework for how to create the Collaboration